src/util/differ.ts

```diff
@@ -0,0 +1,26 @@
+type ObjectJSON = Record<string, unknown>;
+type Field = string | number | ObjectJSON;
+
+class Differ {
+  isEmpty(value: Field): boolean {
+    if (value) {
+      return value === '';
+    }
+    return true;
+  }
+
+  run(requiredFields: ObjectJSON, packageFields: ObjectJSON): string[] {
```

Comment:
nit: we don't need to have `run` method names in utilities classes but that is only for core commands to keep consistency. but I liked the differ.run (I read like --> that is a thing that runs/execute a diff ) :+1: